speakeasy/winenv/api/usermode/kernel32.py

```diff
@@ -2514,7 +2515,7 @@ def GetFullPathName(self, emu, argv, ctx={}):
                     ptr = (lpBuffer + offset).to_bytes(emu.get_ptr_size(), 'little')
                     self.mem_write(lpFilePart, ptr)
 
-                rv = len(fn)
+            rv = len(fn)
```

Comment:
if there's not a buffer, then the number of bytes written would be zero? i dont actually see the MSDN page on this routine so im not sure what the correct behavior is. i'll trust your fix, though.

Reply:
the [MSDN page states](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfullpathnamea#return-value) that the return value contains the number of TCHARs in the returned string. In case of an error the return value is zero except for the case when the destination buffer is not large enough to hold the string. In that case the function is expected to return the amount TCHARs required to hold the string. Since the current implementation does not check the buffer size anyway it seems more reliable to just return the number of bytes in either case if a lpFileName is provided.

Reply:
i promise i know how to use google! weird that it wasn't coming up. thanks for sharing the details. this change makes sense to me.